x-pack/legacy/plugins/maps/public/reducers/map.test.js

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+jest.mock('../actions/map_actions', () => ({}));
+
+import { resetDataRequest } from './map';
+
+describe('reducers/map', () => {
+  it('Should clear datarequest without mutation store state', async () => {
+    const layerId = 'foobar';
+    const requestToken = 'tokenId';
+    const dataId = 'dataId';
+
+    const preState = {
+      layerList: [
+        {
+          id: `not_${layerId}`,
+        },
+        {
+          id: layerId,
+          __dataRequests: [
+            {
+              dataRequestToken: `not_${requestToken}`,
+              dataId: `not_${dataId}`,
+            },
+            {
+              dataRequestToken: requestToken,
+              dataId: dataId,
+            },
+          ],
+        },
+      ],
+    };
+
+    const action = {
+      layerId,
+      requestToken,
+      dataId,
+    };
+
+    const postState = resetDataRequest(preState, action);
```

Comment:
This is a lot of expect statements that masks what you are looking for. You only need to check 2 things
1) preState does not deep equals postState to ensure preState was not changed. This can just be done with `expect(preState).not.toEqual(postState)`
2) postState has set dataRequestToken to null

Reply:
discussed offline. iso comparing post to pre-state, compare copy of pre-state to pre-state.